readthedocs_build/config/test_config.py

```diff
@@ -349,6 +349,193 @@ def it_uses_validate_file(tmpdir):
+            build = get_build_config({
+                'submodules': {
+                    'exclude': ['sub-two'],
+                },
+                'recursive': True,
+            })
+            build.validate_submodules()
+            assert build['submodules']['recursive'] is True
+
+    def it_validates_recursive_key(tmpdir):
+        apply_fs(tmpdir, {'sub-one': {}, 'sub-two': {}})
+        with tmpdir.as_cwd():
+            build = get_build_config({
+                'submodules': {
+                    'exclude': ['sub-two'],
+                },
+                'recursive': 'True',
+            })
+            with raises(InvalidConfig) as excinfo:
+                build.validate_submodules()
+            assert excinfo.value.key == 'submodules.recursive'
+            assert excinfo.value.code == INVALID_BOOL
+
+    def it_accepts_all_keyword_on_include(tmpdir):
+        apply_fs(tmpdir, {'sub-one': {}, 'sub-two': {}})
+        with tmpdir.as_cwd():
+            build = get_build_config({
+                'submodules': {
+                    'include': 'all',
+                },
+                'recursive': True
+            })
+            build.validate_submodules()
+            assert build['submodules']['include'] == ALL
+            assert build['submodules']['recursive'] is True
+
+    def it_validates_the_correct_type_for_include(tmpdir):
+        apply_fs(tmpdir, {'sub-one': {}, 'sub-two': {}})
+        with tmpdir.as_cwd():
+            build = get_build_config({
+                'submodules': {
+                    'include': 'none',
+                },
+                'recursive': True
+            })
+            with raises(InvalidConfig) as excinfo:
+                build.validate_submodules()
+            assert excinfo.value.key == 'submodules.include'
+            assert excinfo.value.code == INVALID_LIST
+
+    def it_validates_the_correct_type_for_exclude(tmpdir):
+        apply_fs(tmpdir, {'sub-one': {}, 'sub-two': {}})
+        with tmpdir.as_cwd():
+            build = get_build_config({
+                'submodules': {
+                    'exclude': 'all',
+                },
+                'recursive': True
+            })
+            with raises(InvalidConfig) as excinfo:
+                build.validate_submodules()
+            assert excinfo.value.key == 'submodules.exclude'
+            assert excinfo.value.code == INVALID_LIST
+
+    def it_parses_explicit_defaults(tmpdir):
+        apply_fs(tmpdir, {'sub-one': {}, 'sub-two': {}})
+        with tmpdir.as_cwd():
+            build = get_build_config({
+                'submodules': {
+                    'include': [],
+                    'recursive': False,
+                }
+            })
+            build.validate_submodules()
+            assert 'submodules' in build
+            assert build['submodules']['include'] == []
+            assert build['submodules']['recursive'] is False
+
+    def it_parses_explicit_wrong_defaults(tmpdir):
```

Comment:
What would be the behavior for this case? Should be an error? Ignore this? or change the value of `recursive` to `False` (this is implemented on this test case)?